jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -935,8 +940,8 @@ ecma_builtin_global_object_decode_uri_helper (ecma_value_t uri __attr_unused___,
         if ((bytes_count == 2 && cp <= LIT_UTF8_1_BYTE_CODE_POINT_MAX)
             || (bytes_count == 3 && cp <= LIT_UTF8_2_BYTE_CODE_POINT_MAX)
             || (bytes_count == 4 && cp <= LIT_UTF8_3_BYTE_CODE_POINT_MAX)
-            || lit_is_code_unit_high_surrogate ((ecma_char_t) cp)
-            || lit_is_code_unit_low_surrogate ((ecma_char_t) cp)
+            || (cp <= LIT_UTF16_CODE_UNIT_MAX && lit_is_code_unit_high_surrogate ((ecma_char_t) cp))
+            || (cp <= LIT_UTF16_CODE_UNIT_MAX && lit_is_code_unit_low_surrogate ((ecma_char_t) cp))
```

Comment:
Seems that checking for equality to high / low surrogate is anyway required. I've added the checks.